src/fty_common_db_asset.cc

```diff
@@ -734,6 +803,77 @@ int select_assets_all_container(tntdb::Connection& conn, std::vector<uint16_t> t
+                    " (((t.id_subtype = " + sensor_sub_id + ") "
+                    "     AND NOT EXISTS "
+                    "         (SELECT * FROM v_bios_asset_element_super_parent AS v WHERE t.id_asset_element=v.id_asset_element "
+                    "         AND v.id_parent1 IS NOT NULL "
+                    "         AND v.id_parent2 IS NOT NULL)) "
+                    // else if device with power chain, parent not defined or power chain not defined
+                    " OR (t.id_subtype IN (" + powerchain_sublist_id + ") "
+                    "     AND "
+                    "         (t.id_parent IS NULL "
+                    "         OR NOT EXISTS "
+                    "             (SELECT id_asset_device_dest FROM t_bios_asset_link_type "
+                    "             AS l JOIN t_bios_asset_link AS a ON a.id_asset_link_type=l.id_asset_link_type "
+                    "             WHERE name=\"power chain\" AND t.id_asset_element=a.id_asset_device_dest))) "
+                    // else for other device with no power chain, parent not defined
+                    " OR (t.id_type IN (" + filter_type_id + ") "
+                    "     AND t.id_subtype NOT IN (" + filter_subtype_id + ") "
+                    "     AND t.id_parent IS NULL)) ";
+
+            // search all device which are configured
+            } else if (configured == "yes") {
+                if (!subtypes.empty() || !types.empty() || status != "" || without != "") {
+                    select += " AND ";
+                }
+                end_select +=
+                    // if device sensor, parent 1 and parent 2 must be configured
+                    " (((t.id_subtype = " + sensor_sub_id + ") "
+                    "     AND EXISTS "
+                    "         (SELECT * FROM v_bios_asset_element_super_parent AS v WHERE t.id_asset_element=v.id_asset_element "
+                    "         AND v.id_parent1 IS NOT NULL "
+                    "         AND v.id_parent2 IS NOT NULL)) "
+                    // else if device with power chain, parent and power chain must be defined
+                    " OR (t.id_subtype IN (" + powerchain_sublist_id + ") "
+                    "     AND "
+                    "         (t.id_parent IS NOT NULL "
+                    "         AND EXISTS "
+                    "             (SELECT id_asset_device_dest FROM t_bios_asset_link_type "
+                    "             AS l JOIN t_bios_asset_link AS a ON a.id_asset_link_type=l.id_asset_link_type "
+                    "             WHERE name=\"power chain\" AND t.id_asset_element=a.id_asset_device_dest))) "
+                    // else for other device with no power chain, parent must be defined
+                    " OR (t.id_type IN (" + filter_type_id + ") "
+                    "     AND t.id_subtype NOT IN (" + filter_subtype_id + ") "
+                    "     AND t.id_parent IS NOT NULL)) ";
+            }
+        }
```

Comment:
add else case to log error "configured malformed"

Reply:
Take into account